### PR TITLE
Phase2-hgx341B Try to provide the dd4hep version of the plugins for passive components of V18 Silicon modules of HGCal

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Try to provide the dd4hep version of the plugins for passive components of V18 Silicon modules of HGCal

#### PR validation:

Pending test using the dumpHGCalPassive in Geometry/HGCalCommonData/test/python

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Review by ianna — COMMENTED** [review]
@bsunanda - from a quick scan I do not see anything wrong with the algorithm. What is the error you get? Thanks!

**Comment by bsunanda** [comment]
@ianna When I run the test code (../src/Geometry/HGCalCommonData/test/python/dumpHGCalPassive_cfg.py type=DD4hep), I get at the end the following message:
DDHGCalPassivePartial: Creating an instance
DDHGCalPassivePartial: Module hgcalPassive:HGCalEEPartialAbsorber1 made of mater
ials:Air T 3.67 Wafer 2r 167.441 Half Separation 0
DD4CMS           ERROR +++ Attempt to access non-existing algorithm option hgcal
:DDHGCalPassivePartial[{lacementIndex]
DD4CMS           ERROR Exception while processing xml source:file:///build/sunanda/CMSSW_13_2_X_2023-06-15-2300/src/Geometry/HGCalCommonData/data/hgcalPassive/v18/hgcalPassive.xml
DD4CMS           ERROR ----> DD4CMS: +++ Attempt to access non-existing algorithm option hgcal:DDHGCalPassivePartial[{lacementIndex]
dd4hep: Error interpreting XML nodes of type <Algorithm/>
dd4hep: Error interpreting XML nodes of type <PosPartSection/>
----- Begin Fatal Exception 16-Jun-2023 03:52:09 CEST-----------------------
An exception of category 'StdException' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'p'
   [2] Prefetching for module DDTestDumpFile/'dump'
   [3] Calling method for EventSetup module DDDetectorESProducer/''
Exception Message:
A std::exception was thrown.
DD4CMS: +++ Attempt to access non-existing algorithm option hgcal:DDHGCalPassivePartial[{lacementIndex]

It does all required steps for the algorithm hgcal:DDHGCalPassiveFull

File: Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalPassivePartial.cc [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    double thick = args.value<double>("ModuleThickness");
+    double waferSize = args.value<double>("WaferSize");
+    double waferSepar = args.value<double>("SensorSeparation");
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalPassivePartial: Module " << parentName << " made of " << material << " T "
+                                  << thick << " Wafer 2r " << waferSize << " Half Separation " << waferSepar;
+#endif
+    std::vector<std::string> tags = args.value<std::vector<std::string>>("Tags");
+    std::vector<int> partialTypes = args.value<std::vector<int>>("PartialTypes");
+    std::vector<int> placementIndex = args.value<std::vector<int>>("{lacementIndex");
```
Comment (ianna) [inline Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalPassivePartial.cc:41]:
```suggestion
    std::vector<int> placementIndex = args.value<std::vector<int>>("PlacementIndex");
```

**Review by ianna — COMMENTED** [review]
@bsunanda - I think, it's just a typo.

File: Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalPassivePartial.cc (hunk 1 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalPassivePartial.cc:41]: Thanks a lot Yana. I could not guess it from the error message.

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by civanch** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

File: Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalPassiveFull.cc [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+#ifdef EDM_ML_DEBUG
+    std::ostringstream st1;
+    for (unsigned int i = 0; i < layerType.size(); ++i)
+      st1 << " [" << i << "] " << layerType[i];
+    edm::LogVerbatim("HGCalGeom") << "There are " << layerType.size() << " blocks" << st1.str();
+
+    edm::LogVerbatim("HGCalGeom") << "==>> Executing DDHGCalPassiveFull...";
+#endif
+
+    static constexpr double tol = 0.00001;
```
Comment (rappoccio) [inline Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalPassiveFull.cc:59]: Any reason why this particular number is chosen? Should we not use numeric_limits instead?

Comment (bsunanda) [reply Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalPassiveFull.cc:59]: Like what? This is typically a tolerance accepted by G4 for overlap testing. If you could provide a better one, we can try

**Pull request merged**